core/themes/basis/templates/comment.tpl.php

```diff
@@ -67,13 +70,18 @@
   <div class="comment-text">
 
     <div class="comment-title">
-      <?php print render($title_prefix); ?>
-      <h3><?php print $title; ?></h3>
-      <?php print render($title_suffix); ?>
+      <?php if ($title_display == TRUE): ?>
+        <?php print render($title_prefix); ?>
+        <h3><?php print $title; ?></h3>
+        <?php print render($title_suffix); ?>
+      <?php endif; ?>
       <?php if ($new): ?>
         <span class="marker"><?php print $new; ?></span>
       <?php endif; ?>
       <span class="comment-time"><?php print $created; ?></span>
+      <?php if ($title_display == FALSE):  ?>
+        <a class="comment-permalink" href="/<?php print $permalink_path; ?>"></a>
+      <?php endif; ?>
```

Comment:
As above (shortened as per @docwilmot's suggestion and remove hard-coded classes).